ext4: prevent a fs mounted read-only from being resized

A filesystem mounted read-only is not allowed to resized online, because
offline resizing is allowed on a filesystem mounted read-only.

Signed-off-by: Yongqiang Yang <xiaoqiangnk@gmail.com>
diff --git a/fs/ext4/ext4.h b/fs/ext4/ext4.h
index a58754f..78866d9 100644
--- a/fs/ext4/ext4.h
+++ b/fs/ext4/ext4.h
@@ -2279,6 +2279,17 @@ static inline int ext4_resize_begin(struct super_block *sb)
 		return -EPERM;
 
 	/*
+	 * We should not do online-resizing on a filesystem mounted
+	 * readonly, because we can do offline-resizing by resiz2fs
+	 * in user space.
+	 */
+	if (sb->s_flags & MS_RDONLY) {
+		ext4_warning(sb, "The filesystem are mounted read-only, "
+			     "so online resizing is not allowed\n");
+		return -EIO;
+	}
+
+	/*
 	 * We are not allowed to do online-resizing on a filesystem mounted
 	 * with error, because it can destroy the filesystem easily.
 	 */
